src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/hearingsbyhearingtype/HearingsByHearingTypeReport.java

```diff
@@ -37,465 +24,389 @@
-            + ZERO + "|"
-            + ZERO + "|"
-            + ZERO + "|"
-            + ZERO + "|"
-            + ZERO + "|"
-            + ZERO + "|"
-            + ZERO;
-
-    public ListingData processHearingsByHearingTypeRequest(ListingDetails listingDetails,
-                                                           List<SubmitEvent> submitEvents,
-                                                           String listingDateFrom,
-                                                           String listingDateTo) {
-        initReport(listingDetails);
-        this.listingDateFrom = listingDateFrom;
-        this.listingDateTo = listingDateTo;
-        casesExistWithHearingStatusHeard = CollectionUtils.isNotEmpty(getDatesList(submitEvents));
+    public HearingsByHearingTypeReportData generateReport(ReportParams params) {
+
+        var submitEvents = getCases(params);
+        var reportData = initReport(params.getCaseTypeId());
+        this.dateFrom = params.getDateFrom();
+        this.dateTo = params.getDateTo();
         if (CollectionUtils.isNotEmpty(submitEvents)) {
-            executeReport(listingDetails, submitEvents);
+            executeReport(submitEvents, reportData);
         }
+        return reportData;
+    }
 
-        listingDetails.getCaseData().clearReportFields();
-        return listingDetails.getCaseData();
+    private HearingsByHearingTypeReportData initReport(String caseTypeId) {
+        var reportSummaryHdr = new HearingsByHearingTypeReportSummaryHdr();
+        var fields = new ReportFields();
+        initReportFields(fields);
+        reportSummaryHdr.setOffice(UtilHelper.getListingCaseTypeId(caseTypeId));
+        reportSummaryHdr.setFields(fields);
+        return new HearingsByHearingTypeReportData(reportSummaryHdr);
     }
 
-    private void executeReport(ListingDetails listingDetails, List<SubmitEvent> submitEvents) {
-        log.info(String.format("Hearings by hearing type report case type id %s search results: %d",
-                listingDetails.getCaseTypeId(), submitEvents.size()));
-        if (UtilHelper.getListingCaseTypeId(listingDetails.getCaseTypeId()).toLowerCase().contains("scotland")) {
-            costsHearingType = "Expenses/Wasted Costs Hearing";
+    private void initReportFields(ReportFields reportFields) {
+        reportFields.setHearingCount("0");
+        reportFields.setCmCount("0");
+        reportFields.setCostsCount("0");
+        reportFields.setTotal("0");
+        reportFields.setHearingPrelimCount("0");
+        reportFields.setReconsiderCount("0");
+        reportFields.setRemedyCount("0");
+    }
+
+    private boolean isValidHearing(DateListedType dateListedType) {
+        LocalDate listedDate;
+        var startDate = LocalDate.parse(ReportHelper.getFormattedLocalDate(dateFrom));
+        var endDate = LocalDate.parse(ReportHelper.getFormattedLocalDate(dateTo));
+        if (isNullOrEmpty(dateListedType.getListedDate())) {
```

Comment:
This should be the first statement in the method.

Reply:
refactored